fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMvRefreshProcessor.java

```diff
@@ -1056,8 +1056,11 @@ public InsertStmt generateInsertStmt(Set<String> materializedViewPartitions,
                 // set partition names for ref base table
                 Set<String> tablePartitionNames = refTableRefreshPartitions.get(nameTableRelationEntry.getKey());
                 TableRelation tableRelation = nameTableRelationEntry.getValue();
-                tableRelation.setPartitionNames(
-                        new PartitionNames(false, new ArrayList<>(tablePartitionNames)));
+                // external table doesn't support query with partitionNames
```

Comment:
What's the behavior before if set the partition names?

Reply:
ut will fail on externalt table.